### ISSUE TITLE
New check: duplicate types in type hint, or duplicate literals in set

### ISSUE THREAD

**Issue by jakkdl** [PyCQA/flake8-bugbear#371]
```python
a: int | int | float = ...
b: Union[List[int, int]] = ...
c: Optional[int] | int | None = ... # can probably rely on other linters to not have to deal with this one
d: set[int] = {1, 2, 3, 3, 5}

e: Tuple[int, int] = ... # not an error
```
these should pretty much always be typos and/or bad copy-pastes, with very few to no false alarms. Don't think it's super common, but definitely not rare or super esoteric.

The check shouldn't have to be too complicated to implement, for the type hint probably need a whitelist of what types you look in (List/list, Set/set, Dict/dict (with special logic), OrderedDict, Optional, Union, Literal...).

For sets you only need to look inside `set()` (ast.Call) and inside `{}` (ast.Set), and probably only include literal int/float/strings.

**Comment by cooperlees** [comment]
This seems super uncontroversial if we can AST parse this out nicely covering the edge cases you've outlined and we can improve on reports ... I'd be happy to add this.

We can run this over some large type checked repos before releasing too to be extra safe.

**Comment by AlexWaygood** [comment]
FYI we already have a check for duplicate items in a union over at [flake8-pyi](https://github.com/PyCQA/flake8-pyi) (error code Y016). Feel free to take inspiration from our implementation :)

**Comment by FozzieHi** [comment]
I think these should be two different checks, but I'm happy to work on a duplicate items in a set check.

> For sets you only need to look inside set() (ast.Call) and inside {} (ast.Set), and probably only include literal int/float/strings.

What do you mean by checking `ast.Call`? Do we also want to catch these?
`set((1, 2, 3, 3, 5))`
`set([1, 2, 3, 3, 5])`

I'm not entirely sure how common these would be, and it would increase the complexity of the check, but I could be missing another syntax here!

**Comment by jakkdl** [comment]
> I think these should be two different checks, but I'm happy to work on a duplicate items in a set check.
> 
> > For sets you only need to look inside set() (ast.Call) and inside {} (ast.Set), and probably only include literal int/float/strings.
> 
> What do you mean by checking `ast.Call`? Do we also want to catch these?
> `set((1, 2, 3, 3, 5))`
> `set([1, 2, 3, 3, 5])`
> 
> I'm not entirely sure how common these would be, and it would increase the complexity of the check, but I could be missing another syntax here!
> 

oops, I misremembered and thought one could write `set(1, 2, 3)` when writing the OP - but that's very much not valid code - so it should only be for `ast.Set`.
